File: backend/verify_api.py
import urllib.request
import json
import sys

with open("api_result.txt", "w") as f:
    f.write("Testing API endpoint http://localhost:8000/api/v1/software/discovered...\n")
    try:
        with urllib.request.urlopen("http://localhost:8000/api/v1/software/discovered", timeout=5) as response:
            if response.status == 200:
                data = json.load(response)
                f.write(f"✅ Success! API returned {len(data)} software items.\n")
                if len(data) > 0:
                    f.write("First 3 items:\n")
                    f.write(json.dumps(data[:3], indent=2))
                else:
                    f.write("⚠️  Warning: API returned empty list.\n")
            else:
                f.write(f"❌ Error: API returned status {response.status}\n")
    except Exception as e:
        f.write(f"❌ Connection Error: {e}\n")

File: backend/verify_api.js
const http = require('http');
const fs = require('fs');

const options = {
    hostname: 'localhost',
    port: 8000,
    path: '/api/v1/software/discovered',
    method: 'GET',
};

const req = http.request(options, (res) => {
    let data = '';
    res.on('data', (chunk) => {
        data += chunk;
    });
    res.on('end', () => {
        try {
            const jsonData = JSON.parse(data);
            fs.writeFileSync('node_api_result.txt', `✅ Success! API returned ${jsonData.length} items.\nFirst 3 items:\n${JSON.stringify(jsonData.slice(0, 3), null, 2)}`);
        } catch (e) {
            fs.writeFileSync('node_api_result.txt', `⚠️ Response received but not valid JSON. Status: ${res.statusCode}\nData: ${data.substring(0, 200)}...`);
        }
    });
});

req.on('error', (e) => {
    fs.writeFileSync('node_api_result.txt', `❌ Connection Error: ${e.message}`);
});
req.end();
